build/conditional-container-push.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+usage() {
+    echo "Usage: $0 IMAGE_URI" >&2
+    exit 1
+}
+
+## image_exists_in_repo IMAGE_URI
+#
+# Checks whether IMAGE_URI -- e.g. quay.io/app-sre/osd-metrics-exporter:abcd123
+# -- exists in the remote repository.
+# If so, returns success.
+# If the image does not exist, but the query was otherwise successful, returns
+# failure.
+# If the query fails for any reason, prints an error and *exits* nonzero.
+#
+# This function cribbed from:
+# https://github.com/openshift/boilerplate/blob/d2d6640ef28a7ce54ac639db32c825aa4bb492a0/boilerplate/_lib/common.sh#L77-L125
+# ...then adapted to use docker rather than skopeo.
+image_exists_in_repo() {
+    local image_uri=$1
+    local output
+
+    if output=$(docker image pull "${image_uri}" 2>&1); then
+        # The image exists. Sanity check the output.
+        local report
+        if report=$(docker image inspect "${image_uri}" 2>&1); then
```

Comment:
This needs to not glom together stdout and stderr. Recently (!) in jenkins, skopeo has started printing things to stderr when the command succeeds. See https://github.com/openshift/boilerplate/pull/162 for the fix.